### PR TITLE
Add install targets to cmake setup

### PR DESCRIPTION
**Pull request by saraedum**
with these, cppast can be installed with `make install` and such, see https://github.com/conda-forge/staged-recipes/pull/16209.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -107,7 +107,7 @@ set(libclang_source
 
 add_library(cppast ${detail_header} ${header} ${source} ${libclang_source})
 target_compile_features(cppast PUBLIC cxx_std_11)
-target_include_directories(cppast PRIVATE ../include SYSTEM INTERFACE ../include)
```
Comment (saraedum) [inline src/CMakeLists.txt:110]: I am not sure what was the idea here (and I am not a cmake expert) but with `PUBLIC` things install correctly for me.

Comment (foonathan) [reply src/CMakeLists.txt:110]: The effect is as follows: for cppast itself, the include headers are passed via -I, which enables warning in them. However, for users the include headers are passed via -isystem, which disables warnings.

What is the issue if it is kept as-is?

Comment (saraedum) [reply src/CMakeLists.txt:110]: If I remember correctly, they were not included correctly in the install target like this.

Comment (saraedum) [reply src/CMakeLists.txt:110]: Let me check what happens without this.

Comment (saraedum) [reply src/CMakeLists.txt:110]: Ok. This is actually not needed. Apparently I changed something else when I modified this and that made things work.

**Review by foonathan — APPROVED** [review]

**Comment by foonathan** [comment]
Thanks!

**Pull request merged**